sycl/include/CL/sycl/handler.hpp

```diff
@@ -551,6 +551,31 @@ class __SYCL_EXPORT handler {
     }
   }
 
+  /// Verifies the kernel bundle to be used if any is set. This throws a
+  /// sycl::exception with error code errc::kernel_not_supported if the used
+  /// kernel bundle does not contain a suitable device image with the requested
+  /// kernel.
+  ///
+  /// \param KernelName is the name of the SYCL kernel to check that the used
+  ///                   kernel bundle contains.
+  void verifyUsedKernelBundle(const std::string &KernelName) {
+    auto UsedKernelBundleImplPtr =
+        getOrInsertHandlerKernelBundle(/*Insert=*/false);
+    if (!UsedKernelBundleImplPtr)
+      return;
+
+    auto UsedKernelBundle =
+        detail::createSyclObjFromImpl<kernel_bundle<bundle_state::executable>>(
```

Comment:
What guarantees that the kernel_bundle is the the executables state here? What happens if it's not?

Reply:
I have made it state-independent by using the kernel bundle's implementation directly instead of creating the kernel bundle object.